### PR TITLE
[IMPROVED] ObjectStore documentation

### PR DESCRIPTION
**Pull request by piotrpio**
Signed-off-by: Piotr Piotrowski <piotr@synadia.com>

### CODE REVIEW

File: jetstream/README.md [hunk 1]
```diff
@@ -797,6 +797,193 @@ fmt.Println(status.Bytes()) // prints the size of all values in bytes
...
+
+The most basic usage of Object bucket is to create or retrieve a bucket and
+perform basic CRUD operations on objects.
+
+```go
+js, _ := jetstream.New(nc)
+ctx := context.Background()
+
+// Create a new bucket. Bucket name is required and has to be unique within a JetStream account.
+os, _ := js.CreateObjectStore(ctx, jetstream.ObjectStoreConfig{Bucket: "configs"})
```
[the remark this answers is not in the document]
Comment (piotrpio) [reply jetstream/README.md:824]: Sorry but I don't get the question I think, how is it optional here?

[the remark this answers is not in the document]
Comment (ripienaar) [reply jetstream/README.md:824]: Oh, nevermind, I just need more coffee and fewer 4am wakeups, all good!

File: jetstream/object.go [hunk 2]
```diff
@@ -35,180 +35,410 @@ import (
...
+
+		// Replicas is the number of replicas to keep for the object store in
+		// clustered jetstream. Defaults to 1, maximum is 5.
+		Replicas int `json:"num_replicas,omitempty"`
+
+		// Placement is used to declare where the object store should be placed via
+		// tags and/or an explicit cluster name.
+		Placement *Placement `json:"placement,omitempty"`
+
+		// Compression sets the underlying stream compression.
```
Comment (Jarema) [inline jetstream/object.go:286]:
```suggestion
		// Compression enables the underlying stream compression.
```

**Review by Jarema — APPROVED** [review]
LGTM!

**Pull request merged**